Update My Subscriptions screen labels and columns

mcliwanow opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes 20986-gh-Automattic/woocommerce.com .

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Go to /plugins/woocommerce' and run 'pnpm --filter='@woocommerce/plugin-woocommerce' build 
3. Open WP-Admin and navigate to WooCommerce > Extensions > My Subscriptions
4. Confirm that title of the second column was changed to `Expires/Renews on` and thrid one was changed to `Subscription`
5. Confirm that status was moved from the name column to the Subscription column and that in case there's no badge we show:
   - "Active" for subscriptions with autorenewals turned on
   - "Cancelled" for cancelled subscriptions
6. Make sure that you have a subscription that didn't expire yet and is both installed and connected to your store. We're testing this condition https://github.com/woocommerce/woocommerce/blob/4a0e06566f6d4bea5acc7748e47ea9faf1273bd6/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx#L372-L373
7. Click the "Renew" button and confirm that you're redirected to `https://woocommerce.com/my-account/my-subscriptions/?key=<product_key>`
8. Note: if you're not logged in to WCCOM you'll be redirect to dashboard after logging in, which is not a great experience. This is to be fixed on the WooCommerce.com side.
9. Please test around this issue

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

mcliwanow committed:
Remove Marketplace feedback modal which is no longer used

The goal of the modal was not to be shown after 2023 (see https://github.com/woocommerce/woocommerce/pull/39609),
so we can remove it from the codebase now. Especially that it's a bit buggy - snackbar prompting for feedback
is displayed but the modal is not there because of the check for FEEDBACK_DIALOG_CAN_APPEAR.

mcliwanow committed:
Marketplace Subscriptions - rename columns

mcliwanow committed:
Marketplace subscriptions - show status badge in the Subscription column

Additionally, we remove the info icon from the status popover

mcliwanow committed:
Marketplace subscriptions - add button helping to turn renewals on

mcliwanow committed:
changelog

mcliwanow committed:
Add focus handlers for accessibility

mcliwanow committed:
Remove unused variable

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx:
```diff
@@ -126,6 +128,44 @@ function getStatusBadge( subscription: Subscription ): StatusBadge | false {
+					),
+					sharing: (
+						<a
+							href={ MARKETPLACE_COLLABORATION_PATH }
```
Shouldn't this be `MARKETPLACE_SHARING_PATH`?

mcliwanow replied:
Nice catch! I copied it from another place and will fix it there now too 🙌

mcliwanow replied:
Done in https://github.com/woocommerce/woocommerce/pull/49985/commits/692c55300ca5e06c62074c5152d7a7045d1a8cb7

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/status-popover.tsx:
```diff
@@ -14,35 +14,78 @@ export default function StatusPopover( props: {
-			<Icon icon={ info } size={ 16 } />
 			{ props.text }
 			{ shouldShowExplanation() && (
 				<Popover
```
The popover content isn't being read by screen readers, so this info is completely invisible for non-sighted users. It would be good if we can improve this as well, though it might make sense to handle as a separate task. It also depends on whether this is an issue with our code, or with the component itself, which is part of WP Core...

mcliwanow replied:
I agree that it's a separate issue and it seems that move in WP Core to `ariakit` might help making the Popover accessible: https://github.com/WordPress/gutenberg/issues/63697#issuecomment-2236241964

KokkieH reviewed (commented):
Looks good, though I have a question about one of the constants you used. I also found an A11Y issue, though I'm not sure if it's something we can fix on our end.

mcliwanow committed:
Fix sharing path url - My Subscriptions

raicem commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/auto-renew-button.tsx:
```diff
@@ -0,0 +1,36 @@
+
+export default function AutoRenewButton( props: AutoRenewProps ) {
+	function recordTracksEvent() {
+		queueRecordEvent( 'marketplace_autor_renew_button_clicked', {
```
```suggestion
		queueRecordEvent( 'marketplace_auto_renew_button_clicked', {
```

raicem reviewed (approved):
Worked great, thank you @mcliwanow.

raicem committed:
Change the status badge priority for expired

We should show the expired badge first, as it makes more sense with the
primary action.

mcliwanow committed:
fix typo

Co-authored-by: Cem Ünalan <raicem@users.noreply.github.com>

mcliwanow committed:
Merge branch 'trunk' into update/my-subscriptions-screen-labels-and-columns

mcliwanow committed:
lint fix

mcliwanow committed:
Merge branch 'trunk' into update/my-subscriptions-screen-labels-and-columns
